pubsublite/streaming-analytics/src/main/java/examples/PubsubliteToGcs.java

```diff
@@ -114,9 +114,8 @@ public static void main(String[] args) throws InterruptedException {
                 .accumulatingFiredPanes())
         .apply("Write elements to GCS", new WriteOneFilePerWindow(options.getOutput(), numShards));
 
-    // Execute the pipeline with an optional timeout for your local program.
-    // This will not cancel the remote job.
-    pipeline.run().waitUntilFinish(Duration.standardMinutes(10));
+    // Execute the pipeline.
+    pipeline.run();
```

Comment:
Seems like leaving a timeout here might be a good idea?

Reply:
When experimented it yesterday, all jobs with `waitUntilFinish()` failed for flex templates. [Here](https://pantheon.corp.google.com/logs/query?dateRangeEnd=2021-04-27T22:53:15.534Z&dateRangeStart=2021-04-27T22:48:19.009Z&expandAll=false&interval=CUSTOM&project=java-docs-samples-testing&timestamp=2021-04-27T22:53:15.534Z&query=resource.type%3D%22dataflow_step%22%0Aresource.labels.job_id%3D%222021-04-27_15_48_18-9752491772384961993%22%0AlogName%3D(%22projects%2Fjava-docs-samples-testing%2Flogs%2Fdataflow.googleapis.com%252Fjob-message%22%20OR%20%22projects%2Fjava-docs-samples-testing%2Flogs%2Fdataflow.googleapis.com%252Flauncher%22)%0A) is an example.

@davidcavazos has raised this issue with the team building flex template before about supporting waiting, but right now it's not.

Reply:
SGTM, thanks for the context.